test/integration/nested_routes_test.rb

```diff
@@ -0,0 +1,295 @@
+      it 'for resources -> resources' do
+        @inspector.must_match 'users_posts GET, HEAD  /users/:users_id/posts         Nested::Controllers::Users::Posts::Index'
+        @inspector.must_match 'new_users_posts GET, HEAD  /users/:users_id/posts/new     Nested::Controllers::Users::Posts::New'
+        @inspector.must_match 'users_posts POST       /users/:users_id/posts         Nested::Controllers::Users::Posts::Create'
+        @inspector.must_match 'users_posts GET, HEAD  /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Show'
+        @inspector.must_match 'edit_users_posts GET, HEAD  /users/:users_id/posts/:id/edit Nested::Controllers::Users::Posts::Edit'
+        @inspector.must_match 'users_posts PATCH      /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Update'
+        @inspector.must_match 'users_posts DELETE     /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Destroy'
+      end
+
+      it 'for resources -> resource' do
+        @inspector.must_match 'new_users_avatar GET, HEAD  /users/:users_id/avatar/new    Nested::Controllers::Users::Avatar::New'
+        @inspector.must_match 'users_avatar POST       /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Create'
+        @inspector.must_match 'users_avatar GET, HEAD  /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Show'
+        @inspector.must_match 'edit_users_avatar GET, HEAD  /users/:users_id/avatar/edit   Nested::Controllers::Users::Avatar::Edit'
+        @inspector.must_match 'users_avatar PATCH      /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Update'
+        @inspector.must_match 'users_avatar DELETE     /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Destroy'
+      end
+
+      it 'for resource -> resources' do
+        @inspector.must_match 'user_comments GET, HEAD  /user/:user_id/comments        Nested::Controllers::User::Comments::Index'
+        @inspector.must_match 'new_user_comments GET, HEAD  /user/:user_id/comments/new    Nested::Controllers::User::Comments::New'
+        @inspector.must_match 'user_comments POST       /user/:user_id/comments        Nested::Controllers::User::Comments::Create'
+        @inspector.must_match 'user_comments GET, HEAD  /user/:user_id/comments/:id    Nested::Controllers::User::Comments::Show'
+        @inspector.must_match 'edit_user_comments GET, HEAD  /user/:user_id/comments/:id/edit Nested::Controllers::User::Comments::Edit'
+        @inspector.must_match 'user_comments PATCH      /user/:user_id/comments/:id    Nested::Controllers::User::Comments::Update'
+        @inspector.must_match 'user_comments DELETE     /user/:user_id/comments/:id    Nested::Controllers::User::Comments::Destroy'
+      end
+
+      it 'for resource -> resource' do
+        @inspector.must_match 'new_user_api_key GET, HEAD  /user/:user_id/api_key/new     Nested::Controllers::User::ApiKey::New'
```

Comment:
@AlfonsoUceda This is wrong, it should be: `/user/api_key/new`. With `resource` you never have `:user_id` around. That is something that `resources` has (notice the plural).